ForumProject/startups/views.py

```diff
@@ -51,7 +61,31 @@ def create(self, request, *args, **kwargs):
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+    def retrieve(self, request, *args, **kwargs):
+        if request.user.user_info.company_id != kwargs['pk'] and request.user.user_info.role != 'investor':
+            return Response({'error': 'You have no permission to view this startup'}, status=status.HTTP_400_BAD_REQUEST)
+        startup = get_object_or_404(self.queryset, pk=kwargs['pk'])
+        serializer = self.serializer_class(startup)
+        return Response(serializer.data)
 
+    def update(self, request, *args, **kwargs):
+        if request.user.user_info.company_id != kwargs['pk']:
+            return Response({'error': 'You have no permission to update startup'}, status=status.HTTP_400_BAD_REQUEST)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+        return Response(serializer.data)
+
+    def partial_update(self, request, *args, **kwargs):
+        if request.user.user_info.company_id != kwargs['pk']:
+            return Response({'error': 'You have no permission to update startup'}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Consider using more appropriate HTTP status codes for permission errors. Using `status.HTTP_403_FORBIDDEN` instead of `status.HTTP_400_BAD_REQUEST` is more semantically correct for permission-related actions

Reply:
Done.